### ISSUE TITLE
Bug: `Filter` removes ms from DateTime values

### ISSUE THREAD

**Issue by jasonsiders** [jasonsiders/apex-starter-pack#64]
While working on a new [feature](https://github.com/jasonsiders/apxsp-big-logger/issues/12) for `apxsp-big-logger`, I came across this fun issue. The `Filter` class currently removes MS from DateTime object values. 

This bug only appeared when attempting to query for an exact DateTime value (as is necessary with the linked application):
```java
Long loggedAtMs = 1679183361820L;
DateTime loggedAtDt = DateTime.newInstance(loggedAtMs);
String queryString = 'SELECT Message__c FROM Log__b WHERE Logged_At__c = :loggedAtDt LIMIT 1';
System.debug('\n\nStrongly Typed SOQL:\n' + queryString + '\n' + JSON.serialize([
SELECT Message__c FROM Log__b WHERE Logged_At__c = :loggedAtDt LIMIT 1
]) + '\n\n');
> Strongly Typed SOQL:
SELECT Message__c FROM Log__b WHERE Logged_At__c = :loggedAtDt LIMIT 1
[{"attributes":{"type":"apxsp__Log__b","url":"/services/data/v57.0/sobjects/apxsp__Log__b/000000000000000AAA"},"apxsp__Message__c":"This is a test, Jason!","Id":"000000000000000AAA"}]

System.debug('\n\nDynamic SOQL:\n' + queryString + '\n' + JSON.serialize(Database.query(queryString)) + '\n\n');
> Dynamic SOQL:
SELECT Message__c FROM Log__b WHERE Logged_At__c = :loggedAtDt LIMIT 1
[{"attributes":{"type":"apxsp__Log__b","url":"/services/data/v57.0/sobjects/apxsp__Log__b/000000000000000AAA"},"apxsp__Message__c":"This is a test, Jason!","Id":"000000000000000AAA"}]

apxsp.Soql query = apxsp.DB.Soql.newQuery(Log__b.SObjectType)
    ?.deselectId()
    ?.selectFields(Log__b.Message__c)
    ?.whereCriteria(new apxsp.Filter(Log__b.Logged_At__c, apxsp.Filter.EQUALS, loggedAtDt))
    ?.setRowLimit(1);
System.debug('\n\napxsp.SOQL:\n' + query.toString() + '\n' + JSON.serialize(query.run()) + '\n\n');
> apxsp.SOQL:
SELECT apxsp__Message__c FROM apxsp__Log__b WHERE apxsp__Logged_At__c = 2023-03-18T23:49:21Z LIMIT 1
[]

```

Cause: in `Filter.cls`, the query string is produced in part with the help of this method:
```java
private String processDateTime(DateTime value) {
	final String soqlFormat = 'yyyy-MM-dd\'T\'HH:mm:ss\'Z\'';
	return value.formatGmt(soqlFormat);
}
```
this is missing the ms (`SSS`) values; should be ` 'yyyy-MM-dd\'T\'HH:mm:ss:SSS\'Z\''` instead.

**Comment by jasonsiders** [comment]
This solution actually wasn't able to solve it. Adding `SSS` returns an error when running via SOQL (or deserializing into a DateTime of any type).

We were able to solve this problem through the new `Database.queryWithBinds()` method: https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_dynamic_soql.htm

```java
DateTime loggedAt = DateTime.newInstance(loggedAtMs);
query.whereCriteria(new apxsp.Filter(Log__b.Logged_At__c, apxsp.Filter.EQUALS, ':loggedAt'));
query.bind('loggedAt', loggedAt);
```

Implementing this now in `Soql`. Still have to work through some finer details + tests.

**Comment by jasonsiders** [comment]
Merged in #65
